### ISSUE TITLE
Write a caproto integration for Lewis simulators

### ISSUE THREAD

**Issue by danielballan** [caproto/caproto#509]
This is an old idea, but I don't think we have an issue for it. I hope to get around to it myself someday, but if someone else feel inspired, go for it!

Lewis: https://github.com/ess-dmsc/lewis

**Comment by klauer** [comment]
Looks like it'd need a caproto-epics "adapter":  https://github.com/ess-dmsc/lewis/blob/master/src/lewis/adapters/epics.py

**Comment by danielballan** [comment]
Lewis expert @MikeHart85 could tell us for sure. (He explained this to me in person during his job interview but I have forgotten the details since.)

**Comment by MikeHart85** [comment]
Yes, writing an adapter that uses caproto would be the way to do it.

At minimum, there would need to be a CaprotoAdapter and a CaprotoInterface class, to implement `Adapter` and `InterfaceBase`. It would probably be nice to also have `PV` class that has the same interface/behavior as the one in the EPICS (pcaspy) adapter, as used [here](https://github.com/ess-dmsc/lewis/blob/master/src/lewis/devices/chopper/interfaces/epics_interface.py#L57-L87), so that the caproto adapter can be a drop in replacements.

I'm probably best suited to do this, unless someone wants to use this as an exercise to learn about Lewis internals.

**Comment by klauer** [comment]
Would be happy to work with you on implementing this from the caproto side, @MikeHart85.
